nf_requirements/v1alpha1/nf_topology_types.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2023 The Nephio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	//apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// NFType defines the type of network functions
+type NFType string
+
+const (
+	NFTypeUPF NFType = "upf"
+	NFTypeSMF NFType = "smf"
+	NFTypeAMF NFType = "amf"
+)
+
+// NFAttachment defines the specification of network attachment points of a NF
+type NFAttachment struct {
```

Comment:
we changed this to interface in the blueprints as people were confused with NADs.

Reply:
OK --- will change